src/algebra/category/Module/change_of_rings.lean

```diff
@@ -0,0 +1,68 @@
+/-
+Copyright (c) 2022 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+import algebra.category.Module.basic
+
+/-!
+# Change Of Rings
+
+## Main definitions
+
+* `category_theory.Module.restrict_scalars`: given rings `R, S` and a ring homomorphism `R ⟶ S`,
+  then `restrict_scalars : Module S ⥤ Module R` is defined by `M ↦ M` where `M : S-module` is seen
+  as `R-module` by `r • m := f r • m` and `S`-linear map `l : M ⟶ M'` is `R`-linear as well.
+-/
+
+
+namespace category_theory.Module
+
+universes u₁ u₂ v
```

Comment:
Since u₁ and u₂ can be inferred from `f`, we want to be able to write `Module.restrict_scalars.{v} f`.
```suggestion
universes v u₁ u₂
```